src/Models/Revision.php

```diff
@@ -71,6 +105,15 @@ class Revision extends MorphPivot
         'metadata'
     ];
 
+    /**
+     * The attributes that should be cast to native types.
+     *
+     * @var array
+     */
+    protected $casts = [
+        'metadata' => 'array'
+    ];
```

Comment:
Was originally going to ask why you removed the json_decode from the meta mutator, until i saw this. Now i have a new question though, don't we have JSON stored in the metadata field, or am i mis-remembering?

Reply:
Actually circling back to this after testing, it looks like the metadata gets stored as `"[]"` / interpreted as a string; can you check that out and see if you're getting the same thing? 
ie: `Post::withoutGlobalScopes()->find(2)->metadata` where id 2 is a post that was just revisioned gives me that string output from above, and not an empty array like id expect

Reply:
Maybe we should add some tests for the metadata storage trait. I originally assumed that by casting it to array I would essentially skip the json_decode step. Let me look into this one.

Reply:
@m-triassi should be resolved now.. also has a test to accompany it

Reply:
Good stuff looks like it was resolved by 106824293cfaf3c0e01661bcd03aa0bd49fee6de!